### PR TITLE
Enable anti-alias font smoothing for all.

### PR DESCRIPTION
**Pull request by yansern**
## Changes proposed in this Pull Request

* Enable anti-aliasing for all WP users.
* Enable anti-aliasing only on HiDPI displays (>= 1.25 / 120dpi).
* Disable anti-aliasing on network admin. p58i-aHm-p2#comment-50960.

**Other PRs:**
* ~PR to enable anti-aliasing on Atomic sites at 710-gh-Automattic/wpcomsh.~ Not needed anymore since it is now `true` by default https://github.com/Automattic/wp-calypso/pull/54196/commits/a1fe347df6873d30c1919cbc110957ebfb3bfce1.
* Phab D63684-code can be merged after deploying this PR to remove unused filter.

See relevant P2 at p58i-aHm-p2.

### Setup

1. Run `install-plugin.sh etk update/enable-font-smoothing-for-all` in your sandbox.

2. Run `install-plugin.sh notifications update/enable-font-smoothing-for-all` in your sandbox.

3. Sandbox the following:
    - your_sandboxed_site.wordpress.com (for `/wp-admin` links)
    - wordpress.com (for `/wp-admin/network` links)
    - widgets.wp.com (for standalone notifications & block editor)
    - public-api.wordpress.com (needed for auth with standalone notifications)

### Test for existence for anti-aliasing class.
  
1. Go to the following url as a regular non-A8C user, the `font-smoothing-antialiased` class should be on the `<body>` tag.
    - http://calypso.localhost:3000/home/your_sandboxed_site.wordpress.com
    - http://calypso.localhost:3000/read
    - http://calypso.localhost:3000/new
    - http://your_sandboxed_site.wordpress.com/wp-admin
    - http://your_sandboxed_site.wordpress.com/wp-admin/post-new.php

2. Go to the following url as a logged-out user, the `font-smoothing-antialiased` class should be on the `<body>` tag.
    - http://calypso.localhost:3000/log-in
  
3. Go to the following url as an A8C user, the `font-smoothing-antialiased` class **should NOT be** on the `<body>` tag.
    - https://wordpress.com/wp-admin/network
 
4. Go to the following url as a regular non-A8C user, the `font-smoothing-antialiased` class should be on the `<body>` tag.
    - https://widgets.wp.com/notifications

### Test between retina & non-retina displays

1. Test any of the above links on an external monitor with non-retina display (below 120dpi).

2. The anti-aliasing effect should not apply.

Related to https://github.com/Automattic/wp-calypso/pull/52230

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/common/index.php [hunk 1]
```diff
@@ -91,7 +91,8 @@ function use_font_smooth_antialiased() {
 		return true;
 	}
 
-	return apply_filters( 'a8c_use_font_smoothing_antialiased', false );
+	// Open anti-aliasing for all by default.
+	return apply_filters( 'a8c_use_font_smoothing_antialiased', true );
```
Comment (p-jackson) [inline apps/editing-toolkit/editing-toolkit-plugin/common/index.php]: Now that we don't need access to is-a11n logic I think we should simplify things and just remove this filter. Doesn't have to be this PR.

File: apps/editing-toolkit/editing-toolkit-plugin/common/index.php [hunk 2]
```diff
@@ -124,7 +125,7 @@ function admin_body_classes( $classes ) {
 		$classes .= ' slider-width-workaround';
 	}
 
-	if ( use_font_smooth_antialiased() ) {
+	if ( use_font_smooth_antialiased() && ! is_network_admin() ) {
```
Comment (p-jackson) [inline apps/editing-toolkit/editing-toolkit-plugin/common/index.php:127]: TIL network admin is a core feature of WP multi-sites, not a wpcom only thing. That means `is_network_admin` will be available on atomic.

File: apps/notifications/src/panel/boot/stylesheets/main.scss [hunk 3]
```diff
@@ -14,10 +14,12 @@
 	// Fixes font anti-aliasing in iframes: andrewmoreton.co.uk/typekit-iframes-safari-weird-antialiasing/
 	-webkit-font-smoothing: subpixel-antialiased;
 
-	body.font-smoothing-antialiased & {
-		text-rendering: optimizeLegibility;
-		-moz-osx-font-smoothing: grayscale;
-		-webkit-font-smoothing: antialiased;
+	@media ( -webkit-min-device-pixel-ratio: 1.25 ), ( min-resolution: 120dpi ) {
```
Comment (p-jackson) [inline apps/notifications/src/panel/boot/stylesheets/main.scss:17]: Very cool way to only enable for retina displays 👍

File: client/landing/gutenboarding/gutenboard.tsx [hunk 4]
```diff
@@ -46,8 +43,7 @@ const Gutenboard: React.FunctionComponent = () => {
 	const { showSignupDialog, onSignupDialogClose } = useSignup();
 	const effectiveFontPairings = useFontPairings();
 
-	const isA8cTeamMember = useSelect( ( select ) => select( READER_STORE ).isA8cTeamMember() );
-	useBodyClass( 'font-smoothing-antialiased', isA8cTeamMember );
+	window.document.body.classList.add( 'font-smoothing-antialiased' );
```
Comment (p-jackson) [inline client/landing/gutenboarding/gutenboard.tsx]: I don't think it's good practice to run side-effects directly inside the body of the component. `useBodyClass()` is still good for this case. But if you think it's overkill (there's nothing async going on so maybe it is), then you can just wrap in a `useEffect()` with `[]` for the dependency list.

**Review by p-jackson — CHANGES_REQUESTED** [review]
> [The wpcomsh change is] not needed since ETK is modified to load for all.

Great! That simplifies things. Glad we can delete a bunch of the code needed to test this out. I'm glad we rolled it out this way though, otherwise we wouldn't have learnt what it looks like on non-retina displays.

Another issue not listed in the comments below is that **the ETK CSS doesn't have the DPI media query**. That's the main issue blocking merge, everything else tested correctly.

File: apps/editing-toolkit/editing-toolkit-plugin/common/index.php (hunk 1 above)
Comment (yansern) [reply apps/editing-toolkit/editing-toolkit-plugin/common/index.php]: Was second guessing whether to simplify this further, thanks for the vote of confidence. https://github.com/Automattic/wp-calypso/pull/54196/commits/20812b72ce67c7b309e9af8ccdfc62245d3c2eac

File: client/landing/gutenboarding/gutenboard.tsx (hunk 4 above)
Comment (yansern) [reply client/landing/gutenboarding/gutenboard.tsx]: Going with `useEffect()` with `[]`. https://github.com/Automattic/wp-calypso/pull/54196/commits/8206296cd90e41e54362029127d5bee4cc21a201

**Comment by yansern** [comment]
> Another issue not listed in the comments below is that **the ETK CSS doesn't have the DPI media query**. That's the main issue blocking merge, everything else tested correctly.

This is now addressed here. Thanks for the review! https://github.com/Automattic/wp-calypso/pull/54196/commits/1866002f13e70bb618e370edd5e9a542ed227144

**Comment by p-jackson** [comment]
Looks like there's a build error in the scss

```
[build:common                        ] ERROR in ./editing-toolkit-plugin/common/index.scss
[build:common                        ] Module build failed (from ../../node_modules/mini-css-extract-plugin/dist/loader.js):
[build:common                        ] ModuleBuildError: Module build failed (from ../../packages/calypso-build/node_modules/sass-loader/dist/cjs.js):
[build:common                        ] SassError: From line 15, column 1 of editing-toolkit-plugin/common/index.scss: 
[build:common                        ]    ╷
[build:common                        ] 15 │ %font-smoothing-antialiased {
[build:common                        ]    │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[build:common                        ]    ╵
[build:common                        ] You may not @extend selectors across media queries.
[build:common                        ]    ╷
[build:common                        ] 23 │         @extend %font-smoothing-antialiased;
[build:common                        ]    │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[build:common                        ]    ╵
```

**Comment by yansern** [comment]
> Looks like there's a build error in the scss
> 
> ```
> [build:common                        ] ERROR in ./editing-toolkit-plugin/common/index.scss
> [build:common                        ] Module build failed (from ../../node_modules/mini-css-extract-plugin/dist/loader.js):
> [build:common                        ] ModuleBuildError: Module build failed (from ../../packages/calypso-build/node_modules/sass-loader/dist/cjs.js):
> [build:common                        ] SassError: From line 15, column 1 of editing-toolkit-plugin/common/index.scss: 
> [build:common                        ]    ╷
> [build:common                        ] 15 │ %font-smoothing-antialiased {
> [build:common                        ]    │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> [build:common                        ]    ╵
> [build:common                        ] You may not @extend selectors across media queries.
> [build:common                        ]    ╷
> [build:common                        ] 23 │         @extend %font-smoothing-antialiased;
> [build:common                        ]    │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> [build:common                        ]    ╵
> ```

Yikes! Checking!

**Comment by yansern** [comment]
Going back to using `@mixin`. https://github.com/Automattic/wp-calypso/pull/54196/commits/b53046ccb0d908e164a968120783348887a5fccb

**Review by p-jackson — APPROVED** [review]
Great!

Screenshot 2021-07-06 at 12 22 20 PM

🥳

**Pull request merged**